dotCMS/src/main/java/com/dotmarketing/portlets/rules/util/RulePermissionableUtil.java

```diff
@@ -40,14 +40,18 @@ public static Permissionable findParentPermissionable(String parent) throws DotD
                 }else{
                     Contentlet contentlet;
 
-                    List<Contentlet> results = APILocator.getContentletAPI()
-                            .search("+identifier:" + parent + " +working:true", 1, 0, null, systemUser, false);
-                    if (!results.isEmpty()) {
-                        contentlet = results.get(0);
-                    } else {
-                        contentlet = APILocator.getContentletAPI().findContentletByIdentifier(parent, false,
-                                    APILocator.getLanguageAPI().getDefaultLanguage().getId(), systemUser, false);
+                    contentlet = APILocator.getContentletAPI().findContentletByIdentifier(parent, false,
+                            APILocator.getLanguageAPI().getDefaultLanguage().getId(), systemUser, false);
+
+                    if (contentlet == null) {
+                        List<Contentlet> results = APILocator.getContentletAPI()
+                                .search("+identifier:" + parent + " +working:true", 1, 0, null,
+                                        systemUser, false);
+                        if (!results.isEmpty()) {
+                            contentlet = results.get(0);
+                        }
                     }
+
                     if (contentlet.isHost()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "contentlet" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)